File: Solutions/74. Search a 2D Matrix/74. Search a 2D Matrix.java
class Solution {
    public boolean searchMatrix(int[][] mat, int target) {
        int m=mat.length,n=mat[0].length;
        int l=0,r=m*n-1;
        while(l<=r){
            int mid=(l+r)/2;
            if(mat[mid/n][mid%n]==target)
                return true;
            if(mat[mid/n][mid%n]<target)
                l=mid+1;
            else
                r=mid-1;
        }
        return false;
    }
}

File: Solutions/74. Search a 2D Matrix/74. Search a 2D Matrix.go
func searchMatrix(mat [][]int, target int) bool {
    m,n:=len(mat),len(mat[0])
    lo,hi:=0,m*n-1
    for lo<=hi{
        mid:=(lo+hi)/2
        if mat[mid/n][mid%n]==target{
            return true
        }
        if mat[mid/n][mid%n]<target{
            lo=mid+1
        }else{
            hi=mid-1
        }
    }
    return false
}